### ISSUE TITLE
parse failure on ars-grin

### ISSUE THREAD

**Issue by jhpoelen** [jhpoelen/samara#27]
as reported by @cmungall from https://build.berkeleybop.org/job/extract-grin-traits/3/console 

```
[https://npgsweb.ars-grin.gov/gringlobal/AccessionObservation.aspx?id=1180774] downloaded.
[https://npgsweb.ars-grin.gov/gringlobal/AccessionDetail.aspx?id=1180776] downloading ...
[https://npgsweb.ars-grin.gov/gringlobal/AccessionDetail.aspx?id=1180776] downloaded.
Exception in thread "main" java.util.NoSuchElementException
	at java.util.ArrayList$Itr.next(ArrayList.java:854)
	at scala.collection.convert.Wrappers$JIteratorWrapper.next(Wrappers.scala:43)
	at scala.collection.Iterator$$anon$11.next(Iterator.scala:409)
	at scala.collection.IterableLike$class.head(IterableLike.scala:107)
	at net.ruippeixotog.scalascraper.model.LazyElementQuery.head(ElementQuery.scala:40)
	at net.ruippeixotog.scalascraper.scraper.ContentExtractors$$anonfun$4.apply(HtmlExtractor.scala:89)
	at net.ruippeixotog.scalascraper.scraper.ContentExtractors$$anonfun$4.apply(HtmlExtractor.scala:89)
	at net.ruippeixotog.scalascraper.scraper.SimpleExtractor.extract(HtmlExtractor.scala:63)
	at net.ruippeixotog.scalascraper.dsl.ScrapingOps$ElementsScrapingOps$$anonfun$extract$1.apply(ScrapingOps.scala:16)
	at scalaz.Monad$$anonfun$map$1$$anonfun$apply$2.apply(Monad.scala:14)
	at scalaz.IdInstances$$anon$1.point(Id.scala:20)
	at scalaz.Monad$$anonfun$map$1.apply(Monad.scala:14)
	at scalaz.IdInstances$$anon$1.bind(Id.scala:22)
	at scalaz.Monad$class.map(Monad.scala:14)
	at scalaz.IdInstances$$anon$1.map(Id.scala:19)
	at scalaz.syntax.FunctorOps.map(FunctorSyntax.scala:10)
	at net.ruippeixotog.scalascraper.dsl.ScrapingOps$ElementsScrapingOps.extract(ScrapingOps.scala:16)
	at net.ruippeixotog.scalascraper.dsl.ScrapingOps$ElementsScrapingOps.$greater$greater(ScrapingOps.scala:20)
	at org.planteome.samara.ParserGrin.parseTaxonInAccessionDetails(ParserGrin.scala:75)
	at org.planteome.samara.ScraperGrin$.getObservationsForAccession(ScraperGrin.scala:58)
	at org.planteome.samara.ScraperGrin$$anonfun$scrape$1$$anonfun$apply$1.apply$mcVI$sp(ScraperGrin.scala:16)
	at org.planteome.samara.ScraperGrin$$anonfun$scrape$1$$anonfun$apply$1.apply(ScraperGrin.scala:15)
	at org.planteome.samara.ScraperGrin$$anonfun$scrape$1$$anonfun$apply$1.apply(ScraperGrin.scala:15)
	at scala.collection.immutable.List.foreach(List.scala:381)
	at org.planteome.samara.ScraperGrin$$anonfun$scrape$1.apply(ScraperGrin.scala:15)
	at org.planteome.samara.ScraperGrin$$anonfun$scrape$1.apply(ScraperGrin.scala:13)
	at scala.collection.immutable.List.foreach(List.scala:381)
	at org.planteome.samara.ScraperGrin$.scrape(ScraperGrin.scala:13)
	at org.planteome.samara.Samara$.delayedEndpoint$org$planteome$samara$Samara$1(Samara.scala:42)
	at org.planteome.samara.Samara$delayedInit$body.apply(Samara.scala:11)
	at scala.Function0$class.apply$mcV$sp(Function0.scala:34)
	at scala.runtime.AbstractFunction0.apply$mcV$sp(AbstractFunction0.scala:12)
	at scala.App$$anonfun$main$1.apply(App.scala:76)
	at scala.App$$anonfun$main$1.apply(App.scala:76)
	at scala.collection.immutable.List.foreach(List.scala:381)
	at scala.collection.generic.TraversableForwarder$class.foreach(TraversableForwarder.scala:35)
	at scala.App$class.main(App.scala:76)
	at org.planteome.samara.Samara$.main(Samara.scala:11)
	at org.planteome.samara.Samara.main(Samara.scala)
Build step 'Execute shell' marked build as failure
Archiving artifacts
```

**Comment by jhpoelen** [comment]
@cmungall @austinmeier I attempted to reproduce this issue. First, I downloaded a copy of the offending page (https://npgsweb.ars-grin.gov/gringlobal/AccessionDetail.aspx?id=1180776) and implemented a test. This test was not able to reproduce the issue. Then, I created a malformed page with missing accession id and associated test and was able to reproduce the issue.

So, it appears that the accession detail pages are malformed every once in while. I've implemented a simple validation mechanism to avoid scraping a page that does not have specific elements in it. Suggest to restart on master branch to verify that this resolved the issue.
